client/src/CategoryList.js

```diff
@@ -0,0 +1,39 @@
+import React, { useEffect, useContext } from 'react'
+import { Card, CardGroup, Button } from 'react-bootstrap'
+import CategoriesContext from './CategoryContext'
+
+const CategoryList = () => {
+  const { categories, setCategories } = useContext(CategoriesContext)
+
+  useEffect(() => {
+    const getCategories = async () => {
+      const response = await fetch(`/api/categories`)
+      const data = await response.json()
+      setCategories(data)
+      console.log(data)
+    }
+    getCategories()
+  }, [])
+
+  return (
+    <div className='categories'>
+      <CardGroup>
+        {categories && categories.map((category) => (
+           <Card style={{ width: '18rem' }} key={category.category_id}>
+             <Card.Img variant='top' src='https://via.placeholder.com/150' />
+             <Card.Body>
+               <Card.Title>
+                 {category.name}
+               </Card.Title>
+               <Button variant='info'>
+                 Events
+               </Button>
+             </Card.Body>
+           </Card>
+         ))}
+      </CardGroup>
```

Comment:
This is all already part of the `CategoryCard` component! I would say to first import `CategoryCard` into `CategoryList`, and inside that map function, use the component and pass in `categories` as a prop.

Reply:
This is good, I would say the only thing additional we need is is `isLoading` and `setIsLoading`  for when we are making a request, it helps to let the user know something is being loaded while we are making a request